frontend/src/metabase/actions/components/ActionViz/Action.unit.spec.tsx

```diff
@@ -76,22 +70,28 @@ function createMockActionDashboardCard(
 }
 
 type SetupOpts = Partial<ActionProps> & {
-  awaitLoading?: boolean;
+  hasDataPermissions?: boolean;
 };
 
 async function setup({
   dashboard = createMockDashboard({ id: DASHBOARD_ID }),
   dashcard = createMockActionDashboardCard(),
   settings = {},
   parameterValues = {},
-  awaitLoading = true,
+  hasDataPermissions = true,
   ...props
 }: SetupOpts = {}) {
-  setupDatabasesEndpoints([DATABASE, DATABASE_WITHOUT_ACTIONS]);
-
-  fetchMock.post(ACTION_EXEC_MOCK_PATH, {
-    "rows-updated": [1],
-  });
+  if (hasDataPermissions) {
+    setupDatabasesEndpoints([DATABASE, DATABASE_WITHOUT_ACTIONS]);
+    fetchMock.post(ACTION_EXEC_MOCK_PATH, { "rows-updated": [1] });
+  } else {
+    const error = {
```

Comment:
WDYT about moving this to `server-mocks`? Like this https://github.com/metabase/metabase/blob/6e17ce37d2dd739d3d5e1b493d60a8805160e516/frontend/test/__support__/server-mocks/card.ts#L33